typing/rec_check.ml

```diff
@@ -577,15 +577,30 @@ let rec expression : Typedtree.expression -> term_judg =
       *)
       expression arg << Guard
     | Texp_apply (e, args)  ->
+        (* [args] may contain omitted arguments, corresponding to labels in
+           the function's type that were not passed in the actual application.
+           The arguments before the first omitted argument are passed to the
+           function immediately, so they are dereferenced. The arguments after
+           the first omitted one are stored in a closure, so guarded.
+           The function itself is called immediately (dereferenced) if there
+           is at least one argument before the first omitted one, stored in
```

Comment:
Nitpick: I find ", stored in the closure otherwise" a bit hard to parse (it is not clear that it is an alternative to "called immediately". I would stop the sentence at the comma and write a new sentence for the second case.